thirdparty/DiskANN/src/pq_flash_index.cpp

```diff
@@ -287,6 +251,37 @@ namespace diskann {
     LOG_KNOWHERE_DEBUG_ << "done.";
   }
 
+  template<typename T>
+  void PQFlashIndex<T>::load_cache_list_from_file(
+      const std::string &cached_nodes_file) {
+    LOG_KNOWHERE_DEBUG_
+        << "Loading the cache list from cached nodes file to memory...";
+    std::vector<uint32_t> node_list;
+    unsigned              cached_nodes_num;
+
+    load_cached_nodes_from_file(aligned_dim, max_degree, cached_nodes_num,
+                                cached_nodes_file, node_list, nhood_cache_buf,
+                                coord_cache_buf);
+    if (cached_nodes_num != node_list.size()) {
```

Comment:
should we check the `cached_nodes_num` with the one calculated from `prep_conf.search_cache_budget_gb`? Or do we really need this field in the `prep_config` at all? We might need it if the users want to reduce the budget, so maybe we can accept any cache size less than the built one? Otherwise, they can delete the file and recalculate how many nodes to cache and regenerate the cached file.

Reply:
According to the comments above